app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+                            break;
+                        }
+                        case ARCHIVED:
+                        default:
+                            archivedPage.current += 1;
+                            nextArchived.current = Boolean(data?.length);
+                            if (data?.length) {
+                                setArchivedChannels([...archivedChannels, ...data]);
+                            }
+                            setLoading(false);
+
+                            break;
+                    }
+                }
+            };
+            switch (typeOfChannels) {
+                case PUBLIC:
+                    if (nextPublic.current) {
+                        setLoading(true);
+                        fetchChannels(
+                            serverUrl,
+                            currentTeamId,
+                            publicPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || [])); // Handle error?
+                    }
+                    break;
+                case SHARED:
+                    if (nextShared.current) {
+                        setLoading(true);
+                        fetchSharedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            sharedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+                case ARCHIVED:
+                default:
+                    if (canShowArchivedChannels && nextArchived.current) {
+                        setLoading(true);
+                        fetchArchivedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            archivedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+            }
+        }
+    }, [loading, typeOfChannels, joinedChannels, archivedChannels, channels, sharedChannels, serverUrl, currentTeamId]);
+
+    const selectActiveChannels = useCallback(() => {
+        switch (typeOfChannels) {
+            case ARCHIVED:
+                return archivedChannels;
+            case SHARED:
+                return sharedChannels;
+            default:
+                return channels;
+        }
+    }, [archivedChannels, channels, sharedChannels, typeOfChannels]);
+
+    const stopSearch = useCallback(() => {
+        setVisibleChannels(selectActiveChannels());
+        setTerm('');
+    }, [selectActiveChannels]);
+
+    const searchChannels = useCallback((text: string) => {
```

Comment:
two things when it comes to this:
1. We should be querying the server to fetch channels that match the type and term (is this supported by the server?)
2. if we are fetching data from the server we should debounce the calls

Reply:
If we are to trust the API documentation, we have two options to search the channels:
- https://api.mattermost.com/#operation/SearchAllChannels (which does not seem to have yet a implementation in the client files (`app/client/rest`))
- https://api.mattermost.com/#operation/SearchChannels (which according to the documentation, only return the channels you are a member of)

V1 seems to be just filtering from the values we got. I see value on that, since when we scroll to get more values, we get values usable even when we are not searching.

Reply:
True on the other hand if you don't scroll enough to load the channels, you would not be able to find the channel you are looking for until you actually scroll

Reply:
Will test, but my theory is that if the screen shows the whole list, it should be calling the searchChannels, even if there is no scroll action.

Reply:
My theory does not work :sweat_smile: Will keep checking. For the time being, any other feedback should be addressed in the latest commit.